### PR TITLE
Branch add favourite field

### PR DESCRIPTION
**Pull request by WuJinhan1**

### CODE REVIEW

**Review by testing1234567891011121314 — COMMENTED** [review]
LGTM

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -145,4 +158,11 @@ public boolean equals(Object other) {
                 && filteredPersons.equals(otherModelManager.filteredPersons);
     }
 
+    @Override
+    public Optional<Person> getPersonByName(Name name) {
```
Comment (Jalenleeruixian) [inline src/main/java/seedu/address/model/ModelManager.java:162]: Good usage of SLAP on a newly defined method.

File: src/main/java/seedu/address/logic/commands/FavoriteCommand.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    /**
+     * Executes the command to toggle the favorite status of the person identified by the name.
+     *
+     * @param model The model containing the address book data.
+     * @return A CommandResult indicating the outcome of the command execution.
+     * @throws CommandException If the person with the specified name is not found in the address book.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```
[the remark this answers is not in the document]
Comment (publicmain) [reply src/main/java/seedu/address/logic/commands/FavoriteCommand.java:35]: LGTM

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 3]
```diff
@@ -42,59 +44,59 @@ public void toModelType_validPersonDetails_returnsPerson() throws Exception {
...
-                new JsonAdaptedPerson(INVALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_ADDRESS, VALID_TAGS);
+                new JsonAdaptedPerson(INVALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_ADDRESS, VALID_TAGS, VALID_IS_FAVORITE);
         String expectedMessage = Name.MESSAGE_CONSTRAINTS;
         assertThrows(IllegalValueException.class, expectedMessage, person::toModelType);
     }
 
     @Test
     public void toModelType_nullName_throwsIllegalValueException() {
-        JsonAdaptedPerson person = new JsonAdaptedPerson(null, VALID_PHONE, VALID_EMAIL, VALID_ADDRESS, VALID_TAGS);
+        JsonAdaptedPerson person = new JsonAdaptedPerson(null, VALID_PHONE, VALID_EMAIL, VALID_ADDRESS, VALID_TAGS, VALID_IS_FAVORITE);
```
[the remark this answers is not in the document]
Comment (publicmain) [reply src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:54]: LGTM

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 4]
```diff
@@ -28,19 +28,24 @@ class JsonAdaptedPerson {
...
     @JsonCreator
-    public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
-            @JsonProperty("email") String email, @JsonProperty("address") String address,
-            @JsonProperty("tags") List<JsonAdaptedTag> tags) {
+    public JsonAdaptedPerson(@JsonProperty("name") String name,
+                             @JsonProperty("phone") String phone,
+                             @JsonProperty("email") String email,
+                             @JsonProperty("address") String address,
+                             @JsonProperty("tags") List<JsonAdaptedTag> tags,
+                             @JsonProperty("isFavorite") Boolean isFavorite) {
```
[the remark this answers is not in the document]
Comment (publicmain) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java:43]: LGTM

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 5]
```diff
@@ -21,22 +21,22 @@ public static Person[] getSamplePersons() {
         return new Person[] {
             new Person(new Name("Alex Yeoh"), new Phone("87438807"), new Email("alexyeoh@example.com"),
                 new Address("Blk 30 Geylang Street 29, #06-40"),
-                getTagSet("friends")),
+                getTagSet("friends"), false),
```
[the remark this answers is not in the document]
Comment (publicmain) [reply src/main/java/seedu/address/model/util/SampleDataUtil.java:24]: LGTM

File: src/main/java/seedu/address/model/ModelManager.java [hunk 6]
```diff
@@ -128,6 +133,14 @@ public void updateFilteredPersonList(Predicate<Person> predicate) {
         filteredPersons.setPredicate(predicate);
     }
 
+    @Override
+    public List<Person> getManagers() {
+        return addressBook.getPersonList().stream()
```
Comment (W3iZhi) [inline src/main/java/seedu/address/model/ModelManager.java:138]: Good use of functional programming to get the data, clean code

**Pull request merged**